### ISSUE TITLE
RelationRemover not working if using "id" only

### ISSUE THREAD

**Issue by xxRockOnxx** [typeorm/typeorm#4767]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[x] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[ ] `latest`
[ ] `@next`
[x] `0.2.16` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

According to: https://typeorm.io/#/relational-query-builder you can just use `id` instead of an `entity` for `add` or `remove`.

However, only `add` works with using `id` only in the following code:

```
interface Musician {
  id: string;
  bio: string;
  roles: ["vocalist", "guitarist", ...etc];
  genres: ["metal", "blues", ...etc];
}

// Tables created by TypeORM
tbl_musician = [id, bio]
tbl_genre = [id]
tbl_role = [id]

tbl_musician_genre = [musician, genre] // these are IDs
tbl_musician_role = [musician, role] // these are IDs

save(musician: Musician) {
  const { roles, genres, ...rest } = musician;
  const repo = this.manager.getRepository(MusicianEntity);

  const addRoles: string[] = [...omitted];  
  const removeRoles: string[] = [...omitted];

  const addGenres: string[] = [...omitted];
  const removeGenres: string[] = [...omitted];

  await repo.save(rest);

  await Promise.all([
    repo.createQueryBuilder()
      .relation("roles")
      .of(musician)
      .addAndRemove(addRoles, removeRoles),
    repo.createQueryBuilder()
      .relation("genres")
      .of(musician)
      .addAndRemove(addGenres, removeGenres)
  ]);
}
```

What do work is the following **workaround**

```js
  await Promise.all([
    repo.createQueryBuilder()
      .relation("roles")
      .of(musician)
      .addAndRemove(addRoles, removeRoles.map(role => ({ id: role }))),
    repo.createQueryBuilder()
      .relation("genres")
      .of(musician)
      .addAndRemove(addGenres, removeGenres.map(genre => ({ id: genre })))
  ]);
```

## Possible Cause

I tried to follow the typeorm's code thru node debugger and on this part:

https://github.com/typeorm/typeorm/blob/master/src/metadata/ColumnMetadata.ts#L548

`entity` is a string.

and on this part:

https://github.com/typeorm/typeorm/blob/master/src/metadata/ColumnMetadata.ts#L608

is what makes the value `undefined`. it looks like `value = "stringValue"["id"]`

**Comment by iMobs** [comment]
Is this not working with the latest version? This might have been fixed by #4734.

**Comment by xxRockOnxx** [comment]
it is exactly the fix for this. thank you
